### DISCUSSION TITLE
The Pagestyle in first page coming as default when generating an ODT Document.

### DISCUSSION THREAD

**Discussion by LeonardoIPM** [eclipse-birt/birt#1802] [Q&A] [answered]
I can't provide the rptdesign, and it can be an issue with our report, not BIRT, but the following problem is happening:

We have a report with landscape orientation on MasterPage, but when emitting it on ODT format and opening in LibreOffice, the first page comes as default pageStyle in portrait orientation.

When emitting the report in doc/docx everything is fine.
Is this a known bug? Is there a workaround? 

![ErroPrimeiraPaginaOdt](https://github.com/user-attachments/assets/33af978a-8d35-4142-8434-79557802f053)

I can change manually the pagestyle and then the document gets right.

![PageStyle](https://github.com/user-attachments/assets/b9539106-6f0b-437f-8588-e933aa6aeb2f)

The problem happens just in the first page, the other ones come with the right pagestyle

**Comment by LeonardoIPM** [comment]
Found out why this is happening, maybe you can help me how to improve my report.

This is my only table structure, the other elements are all considered hidden.
![Outline](https://github.com/user-attachments/assets/ec06000d-4035-4d74-b161-0e2d62461034)

When generating the report in ODT format, in content.xml inside the ODT, the masterpage is setted in Table1.Table1 element.
![Tabela 1](https://github.com/user-attachments/assets/538d6a60-7bcc-4bba-a82a-db9089db7587)

But in Libreoffice, when I try to edit/exclude Table1.Table1 (Here is Tabela5), it shows me Table2.Table2 (Tabela6). So, the properties setted in Tabela1.Tabela1 are being ignored. If I set manually the masterpage in Table2.Table2, then the formatting is corrected.
![Tabela no libreoffice](https://github.com/user-attachments/assets/acdd3c2d-5f3a-479a-8d05-a7723a0e52db)

I don't know how actually the Table Group works, but Tabela5 seems to be the header and only exists to organize the content of other Tabelas, then the first content result is Tabela6, and how it hasn't a masterpage, then the default is setted...?

Is this a BIRT bug or my report was bad created? (Emitting this report in doc/docx and opening in word works fine).

@edit: How to reproduce the issue:

Create a report -> Insert a table -> Insert a table group in the table -> Delete header, details and footer -> insert a grid in the first cell.
Create a masterpage A4 and landscape orientation.
Verify masterpage isn't applied in first page.
![reproduzir](https://github.com/user-attachments/assets/381a9cec-c8b9-4962-b79e-258e201e7778)

**Reply by speckyspooky** [reply]
A suggestion, give your master page a name and asign your table explicit this master page.
Perhaps this direct asigment can help that the master is used directly on your first page.

**Reply by LeonardoIPM** [reply]
Did it, but not working yet... I'm trying to find a workaround to it

**Reply by LeonardoIPM** [reply]
Ok, what is going on:

When you create a table as first element of the report, but then your first cell is also a table, the ODT document/LibreOffice don't read the first table masterpage style, but the second table's one.

Example:
If your first element is a table and the first cell is also a table/grid/something ODT consider a table, then the style will come from the second table. If the first cell from second table is also a table, then the style will come from the third table and so on.

In the case below, Table1 has as it first cell element "Table2", that in its first cell isn't a table. Then, Libreoffice will take Table2 masterpage to define the document, and totally ignore Table1.
```
                  <office:text>
			<table:table table:style-name="Table1.Table1">
				<table:table-column table:style-name="Table1.Column1"
					table:number-columns-repeated="2" />
				<table:table-column table:style-name="Table1.Column2" />
				<table:table-row>
					<table:table-cell table:style-name="Table1.Cell1">
						<table:table table:style-name="Table2.Table2">
							<table:table-column table:style-name="Table2.Column3"
								table:number-columns-repeated="2" />
							<table:table-column table:style-name="Table2.Column4" />
							<table:table-row>
								<table:table-cell table:style-name="Table2.Cell2">
									<text:p>
										<text:bookmark text:name="__bookmark_1" />
									</text:p>
```

**Comment by LeonardoIPM** [comment, marked as answer]
Well, as a workaround, when we are processing an ODT report, we are placing a label with line height 0 as the first element by DesignHandle. The masterpage is setted in the label and the file can load it.

A bug was registered about it on bugzilla: https://bugs.documentfoundation.org/show_bug.cgi?id=162243
